src/internal/connector/graph_connector_onedrive_test.go

```diff
@@ -278,21 +309,21 @@ type onedriveTest struct {
 	cols         []onedriveColInfo
 }
 
-func testDataForInfo(t *testing.T, cols []onedriveColInfo, backupVersion int) []colInfo {
+func testDataForInfo(t *testing.T, cols []onedriveColInfo, backupVersion int, permUseID bool) []colInfo {
 	var res []colInfo
 
 	for _, c := range cols {
 		onedriveCol := newOneDriveCollection(t, c.pathElements, backupVersion)
 
 		for _, f := range c.files {
-			onedriveCol.withFile(f.name, f.data, f.perms.user, f.perms.roles)
+			onedriveCol.withFile(f.name, f.data, f.perms, permUseID)
```

Comment:
nit: if we're always going to have either all emails or all IDs in the permission info for backups then you can just make `permUseID` a field in the `onedriveCollection` struct like backup version is


edit: actually since it generally seems like it's set based on the backup version you could roll that into the onedriveCollection helper functions and not pass an extra parameter at all

Reply:
> actually since it generally seems like it's set based on the backup version you could roll that into the onedriveCollection helper functions and not pass an extra parameter at all

~I was kinda debating this. This is not really based on version as there is no specific version(no new version created for this change) which switches it from email to id, plus both email and id should work with all version. We probably should run tests with both id and email for all versions but decided to not do that so that to double the test count.~

On second thought, I think it is OK to just base it off of the version.